aws/credentials/endpointcreds/provider.go

```diff
@@ -164,7 +172,22 @@ func (p *Provider) getCredentials(ctx aws.Context) (*getCredentialsOutput, error
 	req := p.Client.NewRequest(op, nil, out)
 	req.SetContext(ctx)
 	req.HTTPRequest.Header.Set("Accept", "application/json")
-	if authToken := p.AuthorizationToken; len(authToken) != 0 {
+
+	authToken := p.AuthorizationToken
+	var err error
+
+	if authFilePath := os.Getenv(httpProviderAuthFileEnvVar); authFilePath != "" {
```

Comment:
This will also facilitate testing the behavior, since the dynamic load behavior for the token can now be plugged into the test implementation.

Reply:
Note that I updated my comment above. `AuthorizationTokenProvider` became an interface in the `endpointcreds` package (instead of an inline function type).